selfie_segmentation/main.js

```diff
@@ -127,6 +144,15 @@ async function renderCamStream() {
 }
 
 async function drawOutput(outputBuffer, srcElement) {
+  if (modelName.startsWith('deeplab')) {
+    // Do additional `argMax` for DeepLabV3 model
+    outputBuffer = tf.tidy(() => {
+      const a = tf.tensor(outputBuffer, [1, 257, 257, 21], 'float32');
+      const b = tf.argMax(a, 3);
```

Comment:
Is this op run on wasm backend? We probably should ensure that to avoid any unnecessary data copying between CPU and GPU.

Reply:
No, it's default webgl backend, and happened in post-processing, does that impact? The post-processing uses WebGL as well.

Reply:
It may impact. If this runs on webgl backend, it means `a` will be uploaded to GPU from CPU and later the `c.dataSync()` will read it back from GPU to CPU. And as you mentioned, if the post-processing it WebGL, it will be uploaded again to GPU. I'd recommend to use tf.js wasm backend to run `tf.argMax` to avoid unnecessary uploading-argMax-downloading. Uploading for WebGL post-processing at last step would be fine.

Reply:
@huningxin, thanks much for your clear explanations! Done.